pkg/prometheus/promcfg.go

```diff
@@ -94,11 +96,28 @@ func NewConfigGenerator(logger log.Logger, p monitoringv1.PrometheusInterface, e
 		return nil, fmt.Errorf("failed to parse scrape classes: %w", err)
 	}
 
+	endpointSliceConfigured := false // Always assume false to preserve original prometheus-operator behaviour.
+
+	// Check if the user has explicitly set the service discovery role to use.
+	switch serviceDiscoveryRole := ptr.Deref(cpf.ServiceDiscoveryRole, monitoringv1.EndpointsRole); serviceDiscoveryRole {
+	case monitoringv1.EndpointSliceRole:
+		level.Info(logger).Log("msg", "using endpointslice as service discovery role")
+		endpointSliceConfigured = true
+	case monitoringv1.EndpointsRole:
+		level.Info(logger).Log("msg", "using endpoints as service discovery role")
+		endpointSliceConfigured = false
```

Comment:
```suggestion
```
This is no-op because the value is already false :)

Reply:
Line 99 is also a no-op as we assign false to the bool `endpointSliceConfigured`.

Perhaps we keep this no-op because I believe being explicit about intention instead of implicit is more helpful to future readers of the code.

Reply:
Fair point :)